### DISCUSSION TITLE
iBFT autoconfiguration enabled by default if iBFT is detected

### DISCUSSION THREAD

**Discussion by netcho** [dracutdevs/dracut#1962] [Ideas]
Hello everyone,

I have stumbled upon a rather interesting situation.

I have a machine booting using iSCSI as a root device on CentOS 7. The initial netboot is handled via iPXE, which attaches the iSCSI disk and loads GRUB. IP configuration is done via DHCP. I have the following extra kernel command line arguments: `rd.iscsi.firmware=1 ip=mgmt0:dhcp`. 
`mgmt0` is the interface that is used for provisioning and network boot. This way, dracut brings up the interface correctly and then attaches the iSCSI root disk, and the OS boots as expected. 

Things get strange when building the kdump initrd image. I'm trying to use the same iSCSI disk for storing the kernel vmcore file. It turns out, however, that when building the kdump initrd image, dracut prepends an `ip=ibft` parameter to the kernel command line via `/etc/cmdline.d/95iscsi.conf`. As a result, the kdump initrd image fails to mount the iSCSI disk correctly and the kdump crash collection fails. 

I've manually patched this function https://github.com/dracutdevs/dracut/blob/4012314f204806a67d038f378f525646814d12ea/modules.d/95iscsi/module-setup.sh#L36 so that it doesn't prepend the `ip=ibft` parameter (it's this one in CentOS 7, I've tried with `rd.iscsi.ibft=1` too) and it worked out as expected.

So, to summarize: Should dracut force iBFT interface configuration when it's not sure how the current OS was booted? IMO, it should look up whether iBFT was used for interface configuration, and if this is the case, set `rd.iscsi.ibft=1` or its older equivalents. If no, it's then up to the user to figure out a way to configure the appropriate network settings.
